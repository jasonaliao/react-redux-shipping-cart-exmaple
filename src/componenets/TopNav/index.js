import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import MiniCart from '../MiniCart';

export default class TopNav extends Component {

    render() {

        return (
            <nav className="top-nav columns">
                <div className="logo-wrap column is-4">
                    <Link to="/" className="logo">React Store</Link>
                </div>
                <div className="column">
                    <MiniCart />
                </div>

            </nav>
        );
    };
}

