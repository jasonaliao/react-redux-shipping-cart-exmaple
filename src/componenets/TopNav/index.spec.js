import React from 'react';
import Enzyme, { shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import TopNav from './'

Enzyme.configure({ adapter: new Adapter() });

test('TopNav component should render as expected', ()=>{
    // snapshots
    const component = shallow(<TopNav />);
    const tree = toJson(component);

    expect(tree).toMatchSnapshot()
});

