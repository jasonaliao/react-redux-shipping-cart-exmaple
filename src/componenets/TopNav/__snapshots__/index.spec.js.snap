// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNav component should render as expected 1`] = `
<nav
  className="top-nav columns"
>
  <div
    className="logo-wrap column is-4"
  >
    <Link
      className="logo"
      replace={false}
      to="/"
    >
      React Store
    </Link>
  </div>
  <div
    className="top-nav-controls column"
  >
    <Connect(MiniCart) />
    <Connect(ProductSearch) />
  </div>
</nav>
`;
